keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepository.kt

```diff
@@ -45,6 +45,23 @@ interface DeliveryConfigRepository : PeriodicallyCheckedRepository<DeliveryConfi
    */
   fun deleteByApplication(application: String): Int
 
+  /**
+   * Deletes a delivery config, and the environments within in it.
```

Comment:
Hmm... I think it would be more intuitive if it was the other way around. I realize this would require `DeliveryConfigRepository` to hold references to `ArtifactRepository` and `ResourceRepository` to do this, but I wondered if that's considered an anti-pattern for some reason? From a logical modeling standpoint, it would seem much easier to wrap my brain around than having the separate `Cleaner` class coordinating this kind of thing.

Reply:
Instead of one repository calling into another, I actually see no reason not to read/write directly to any table from from `DeliveryConfigRepository`. The thing I find useful about our current repository model is in how they enable periodic polling with a distributed lock per item polled. I think it makes sense if that's all a repository was, not an owner of tables.

Reply:
Created a jira to track that larger change, which definitely needs to happen: https://github.com/spinnaker/keel/issues/740